WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -163,9 +163,11 @@ class WordPressAppDelegate: UIResponder, UIApplicationDelegate {
     func applicationDidEnterBackground(_ application: UIApplication) {
         DDLogInfo("\(self) \(#function)")
 
-        // Let the app finish any uploads that are in progress
         let app = UIApplication.shared
+
+        // Let the app finish any uploads that are in progress
         if let task = bgTask, bgTask != .invalid {
+            DDLogInfo("BackgroundTask: ending existing backgroundTask for bgTask = \(task)")
             app.endBackgroundTask(task)
             bgTask = .invalid
         }
```

Comment:
When I was working on this issue + before the PR, I had thought long and hard about this one too as I didn't really understand the purpose of the old-school backgroundtask until I saw the comment on line 168 in `WordPressAppDelegate.swift`

`//Let the app finish any uploads that are in progress`

I assume this meant that media uploads could potentially still be processing and the app is trying to give them as much extra time as possible to upload?  I didn't remove it as such because I wasn't confident I'd have a way to test this + all the potential scenarios it impacts.  Maybe you'd have further thoughts / insight as to if we won't impact anything doing this?

Reply:
I think it seems as though that is the intention of it, yes. I'm not a fan of the system message that states that the app may be killed, however – but I think this is something that we can potentially look at addressing separately.

Just for reference, I took a look back through the Git history and it seems that this background task is _very_ old. The first reference to this comment can be found here: https://github.com/wordpress-mobile/WordPress-iOS/blame/6f29c6507cbd3c790889fbee2752feffbf822385/WordPress/Classes/System/WordPressAppDelegate.m#L152

Now that we have MediaCoordinator and PostCoordinator, I wonder if we can at least restrict this background task to be started only if we know there is already an upload in progress (and potentially to call the completion handler when all open uploads complete).